Feat/rc/instrumentation test

trOnk12 opened the pull request:
----
#### PR Submission Checklist for internal contributors

- The **PR Title**
    - [ ] conforms to the style of semantic commits messages¹ supported in Wire's Github Workflow²
    - [ ] contains a reference JIRA issue number like `SQPIT-764`
    - [ ] answers the question: _If merged, this PR will: ..._ ³

- The **PR Description**
    - [ ] is free of optional paragraphs and you have filled the relevant parts to the best of your ability
----

# What's new in this PR?

### Issues

_Briefly describe the issue you have solved or implemented with this pull request. If the PR contains multiple issues, use a bullet list._

### Causes (Optional)

_Briefly describe the causes behind the issues. This could be helpful to understand the adopted solutions behind some nasty bugs or complex issues._

### Solutions

_Briefly describe the solutions you have implemented for the issues explained above._

### Dependencies (Optional)

_If there are some other pull requests related to this one (e.g. new releases of frameworks), specify them here._

Needs releases with:

- [ ] GitHub link to other pull request

### Testing

#### Test Coverage (Optional)

- [ ] I have added automated test to this contribution

#### How to Test

_Briefly describe how this change was tested and if applicable the exact steps taken to verify that it works as expected._

### Notes (Optional)

_Specify here any other facts that you think are important for this issue._

### Attachments (Optional)

_Attachments like images, videos, etc. (drag and drop in the text box)_ 
<!-- Uncomment the brackets and place your screenshots on the table below. -->
<!--
| Before | After |
| ----------- | ------------ |
|   |   |
-->

----
#### PR Post Submission Checklist for internal contributors (Optional)

- [ ] Wire's Github Workflow has automatically linked the PR to a JIRA issue
----
#### PR Post Merge Checklist for internal contributors

- [ ] If any soft of configuration variable was introduced by this PR, it has been added to the relevant documents and the CI jobs have been updated.
----
##### References
1. https://sparkbox.com/foundry/semantic_commit_messages
1. https://github.com/wireapp/.github#usage
1. E.g. `feat(conversation-list): Sort conversations by most emojis in the title #SQPIT-764`.

deanrobertcook commented on app/src/main/scala/com/waz/zclient/WireApplication.scala:
```diff
@@ -284,8 +286,23 @@ class WireApplication extends MultiDexApplication with WireContext with Injectab
+        sslContext.init(null, null, null)
+        sslContext.createSSLEngine
+      } catch {
+        case error: Throwable =>
```
should probably catch `NonFatal(e)` here

yeryomenkom replied:
Any reasons?

deanrobertcook replied:
Because you don't want to stop the app from crashing if there is an OOO error or an Unsatisfied link error - in fact, you should really figure out which Exceptions these methods throw and only handle those, but at least with NonFatal(e) you're filtering out the really bad things that should crash the app

deanrobertcook commented on app/src/main/java/com/waz/zclient/markdown/MarkdownTextView.java:
```diff
@@ -81,15 +81,15 @@ private void configureStyleSheet() {
+//                setLinkTextColor(color.getColor());
+//                refreshLinks();
+//            }
+//        }, eventContext());
```
Let's fix this properly and in another commit

yeryomenkom replied:
Fix is in https://github.com/wireapp/wire-android/pull/1705
So, we should merge that PR first. After that I will this branch.

yeryomenkom commented:
build again

yamilmedina committed:
chore: bump kalium ref: security banner and devices deletion fix (#1585)

gongracr committed:
fix(rc): wrong German wordings [AR-3239][AR-3240] (#1578)  (#1586)

* fix: wrong German wordings [AR-3239][AR-3240] (#1578)

* fix: wrong German wordings

* fix: detekt issues

* fix: fixed tests

* fix: removed unused methods

saleniuk committed:
fix(rc): showing attachment options on Android 8 and items ripple effect [AR-3033] [AR-3070] (#1589)

* fix: showing attachment options on Android 8, items ripple effect

* fix detekt

saleniuk committed:
fix(rc): message edit issues [AR-3241] (#1577)

* fix: message edit issues [AR-3241]

* review change

saleniuk committed:
refactor(rc): split click blocking flag into two - block when syncing and when connecting [AR-2693] (#1587)

* refactor: add ClickBlockParams - distinguish between syncing and connecting

* update kalium ref

* fix detekt

* update kalium ref

* self review

* trigger build

ohassine committed:
fix: crash when putting an ongoing call in background (#1583) (#1591)

Co-authored-by: Alexandre Ferris <ferris.alexandre@gmail.com>

Garzas committed:
fix: show reply message when quoted message is not available [AR-3238] (#1597)

* fix: show reply message when quoted message is not available

yamilmedina committed:
fix(rc): federated user data is not refreshed (AR-3249) (#1604)

* fix: add checks for details tab

* fix: kalium ref updated

* chore: bump kalium ref federated user that refresh

Garzas committed:
fix: remove read messages after migration [AR-3237] (#1581)

* fix: remove read messages after migration
* fix: clear read events for multi accounts migration

ohassine committed:
fix: crash when putting an ongoing call in background (AR-3260) (RC) (#1601)

* fix: crash when putting an ongoing call in background

* fix: magic number

* fix: unit test

* fix: crash when putting ongoing call in background

ohassine committed:
fix: camera is turned on after leaving and rejoing a call (#1617)

alexandreferris committed:
fix: (RC) personal user settings group read receipt (AR-3168) (#1608)

* chore: update kalium reference

* fix: adjust logic for enabling and disabling toggle for read receipts in group settings

* test: add tests for new logic

yamilmedina committed:
fix(rc): banner padding issue and classified banners for calls (AR-3244) (#1620)

* chore: fix banner padding issue for calls

* chore: detekt

saleniuk committed:
fix: attachment options not appearing on android 8 [AR-3033] (#1613)

yamilmedina committed:
fix(rc): federation classified banner displayed after leaving or user deleted (AR-3264) (#1629)

* fix: add classification banners to disabled input

* fix: add classification banners to disabled input

ohassine committed:
fix: call still ringing despite that the caller already ended the call (AR-3046) (RC) (#1592)

* fix: call still ringing despite that the caller already ended the call (AR-3046) (#1575)

* fix: Call still ringing despite that the caller already ended the call

* fix: detekt

* fix: unit test

* fix: revert timeout

* chore: revert dependency update

* chore: revert dependency update

---------

Co-authored-by: Alexandre Ferris <ferris.alexandre@gmail.com>

tmspzz committed:
chore: update kalium ref

ohassine committed:
fix: ANR caused by workManager (AR-3279) (RC) (#1642)

* chore: update workManager dependency

* chore: update workManager dependency

ohassine committed:
fix: proximity sensor crash (AR-3276)  (#1645)

* fix: proximity sensor crash

* fix: sensor crash

* fix: update kalium reference

Garzas committed:
fix: empty user profle (#1648)

mchenani committed:
fix(conversation): hide conversation protocol info when mls is disabled [RC] (#1633)

* fix(conversation): hide conversation protocol info when mls is disabled

vitorhugods committed:
chore: update kalium reference

ohassine committed:
fix: ForegroundServiceDidNotStartInTimeException (#1657)

vitorhugods committed:
refactor: customization file support (#1652)

MohamadJaara committed:
feat: [RC] crash app on migration erorrs (#1658)

* chore: crash the migration worker if there is an error (#1579)

* chore: crash the migration worker if there is an error

* styling

* trigger CI

* address PR comments

(cherry picked from commit f66e02fdfd9ebe1f1756146afc43aa0b4a0c3f68)

* feat: redo migration on app update when failure (#1584)

* chore: crash the migration worker if there is an error

* feat: redo migration when completed with errors on a previous version

* styling

* trigger CI

* address PR comments

* styling

* unit tests

(cherry picked from commit 172b2bf84ee82b23f9fd5b07bb169f83adcd7404)

* fix: check if user is logged in after migraion error (#1607)

* fix: check if user is logged in after migraion error

* clear backstack when navigating to login

(cherry picked from commit 825f5b086888f9a72e5ad3d660f169675af7ae2c)

* fix: try catch migration errors and include the report when failing (#1616)

Co-authored-by: Tommaso Piazza <196761+tmspzz@users.noreply.github.com>

(cherry picked from commit ed2827f7a2a91f390f764b77fa51e459a97ea815)

* fix: propagate migration errors to the UI and crash in the view model (#1628)

* fix: propagate migration errors to the UI and crash in the view model

* revert disabling strict mode

* remove unnecessary logs

(cherry picked from commit fff30dab75bf0fd4f8bca5e1bb29054583daca58)

* fix: merge issues

* styling

---------

Co-authored-by: Vitor Hugo Schwaab <vitor@schwaab.xyz>

vitorhugods committed:
fix: missing SUPPORT_URL from compatRelease builds

ohassine committed:
chore: disable back button for incoming and outgoing call screens (AR-3298) (RC) (#1669)

* chore: disable back button for incoming and outgoing call screens

* chore: detekt

---------

Co-authored-by: Mohamad Jaara <mohamad.jaara@wire.com>

Garzas committed:
fix: default config for firebaseAppId (#1670)

Garzas committed:
fix: null pointer on mention span (#1672)

tmspzz committed:
chore: update kalium ref

MohamadJaara committed:
feat(build-config): add production build config (RC) (AR-2952) (#1389) (#1677)

* feat(build-config): add production build config (RC) (AR-2952) (#1389)

* feat(build-config): add production build config (RC)

* hide mls options in debug screen when mls is disabled

* apply icon and dependency filtering

* disable minify

* extract datadog code to ExternalLoggerManager

* add missing build flags

* Unit

* detekt

* DataDogLogger

---------

Co-authored-by: Mojtaba Chenani <chenani@outlook.com>

vitorhugods committed:
chore: support building with custom backends (#1675)

MohamadJaara committed:
fix: missing shared user id in the Manifest

tmspzz committed:
chore: update kalium ref

vitorhugods committed:
ci: accept empty git password

vitorhugods committed:
ci: use `custom-reloaded.json` customization file

This is in order to keep compatibility with current Scala Android `custom.json` file

vitorhugods committed:
ci: support screaming snake case environment variables

Garzas committed:
fix: update view model state race condition (#1680)

saleniuk committed:
fix: mentions are lost after edit [AR-3299] (#1697)

trOnk12 committed:
feat(rc): provide ui related custom configuration (#1699)

vitorhugods committed:
chore: cleanup build flags (#1701)

- Unused build flags removed.
- Standardise casing of all remaining.
- Attempt to group into categories.
- Add simple documentation

tmspzz committed:
chore: update kalium ref

tmspzz committed:
chore: update kalium ref

AndroidBob commented:
Build [1](https://10.10.124.134/job/wire-android-reloaded/job/PR-1710/1/) **failed**.
